lib/compat/wordpress-6.6/resolve-patterns.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+/**
+ * Replaces pattern blocks with their content.
+ *
+ * @param array $blocks Array of blocks.
+ * @param array $inner_content Optional array of inner content.
+ * @return array Array of blocks with patterns replaced.
+ */
+function gutenberg_replace_pattern_blocks( $blocks, &$inner_content = null ) {
+	// Keep track of seen references to avoid infinite loops.
+	static $seen_refs = array();
+	$i                = 0;
+	while ( $i < count( $blocks ) ) {
+		if ( 'core/pattern' === $blocks[ $i ]['blockName'] ) {
+			$slug = $blocks[ $i ]['attrs']['slug'];
+
+			if ( isset( $seen_refs[ $slug ] ) ) {
+				// Skip recursive patterns.
+				array_splice( $blocks, $i, 1 );
+				continue;
+			}
+
+			$registry           = WP_Block_Patterns_Registry::get_instance();
+			$pattern            = $registry->get_registered( $slug );
+			$blocks_to_insert   = parse_blocks( $pattern['content'] );
```

Comment:
I'm seeing a PHP warning in WP 6.4.3 while running Gutenberg trunk on this line:

<img width="861" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/dbd4729b-e117-4013-ab74-47c751bb8d53">

Should there be a check that the pattern could be found? I wasn't sure if there's a bug here, or if it's something specific to my testing environment.

Reply:
I had the same issue fwiw, and wp-admin wasn't even loading for me but pretty sure the version I was on was 6.5-alpha-something.

Destroying my env and upgrading to 6.6-alpha fixed it, but we do need to make sure the plugin works on 6.4 and 6.5.

Reply:
Thank, I created https://github.com/WordPress/gutenberg/pull/60464